src/libcmd/meson.build

```diff
@@ -1,7 +1,10 @@
 # libmain build file
 #============================================================================
 
-libcmd_inc = [include_directories('.')]
+libcmd_inc = [include_directories(
+    '.',
+    join_paths('..'),
```

Comment:
this will cause problems with file inclusion, you cannot chain directories like this in meson, and then append extra directories to the include_directories target; thats the reason its referenced as an array.  you should also not use relative paths, its strongly discouraged when creating buildsystem files. please reference the other `meson` files and model your commits based on them.

Reply:
> this will cause problems with file inclusion, you cannot chain directories like this in meson, and then append extra directories to the include_directories target;

I don't see the problem it causes...
This is not chaining directories, this is passing several arguments to `include_directories`.

And BTW, in the documentation, it is said that the `include_directories` parameters used in `library()` **MUST** be an `include_directories` object (not an array!): https://mesonbuild.com/Reference-manual.html#executable (note that https://mesonbuild.com/Reference-manual.html#library 's documentation states "The keyword arguments for this are the same as for executable with the following additions").

> you should also not use relative paths, its strongly discouraged when creating buildsystem files

This is what I get with absolute paths:
```
src/libutil/meson.build:4:0: ERROR: Tried to form an absolute path to a source dir. You should not do that but use relative paths instead.
```

> please reference the other meson files and model your commits based on them

What do you mean by that?

Reply:
once you create an include_directories() target, you cannot append to it. look through the codebase, youll see directories are appended to it per sub directory, instead of at a top level. they are appended to an array which is then passed to the build object; instead of hardcoding all the directories at the top file for that target. using `../` in any capacity causes massive headaches in any buildsystem, and ive lost many hours of work fixing relative paths in MSVC and CMake buildsystems. there is no need to use them, especially in the nix-meson buildsystem, as the .. directory already has a include_directories() target. just reference the already existing target, theres no need to do any kind of path traversal.

Reply:
this also breaks header inclusion for build targets. instead of properly inheriting the headers from the build target, it fetches them from the top level src directory. this can result in installing the same headers multiple times in different locations as instead of sharing the same headers, each library installs a set of the same headers of its own.